### DISCUSSION TITLE
Including multiple container files containing relationships (in model)

### DISCUSSION THREAD

**Discussion by okamiraner** [structurizr/cli#139] [Q&A] [answered]
Hello

I am trying to place all containers in a folder in separate files and then include all folder contents in a software system. Each container DSL has info on outbound relationships in it and there is the problem (in order to avoid conflicts and a large confusing list of relationships). When loading a single file, the exporter ignores ones referenced in relationships and returns the following error:

**Command:**

```
$ structurizr-cli export -w <src>.dsl -f mermaid -o <trg>

The destination element "<target_container>" does not exist at line <line>: -> <target_container> "" "HTTP" ...`
```

**Folder tree:**

```
...
├── model
│   ├── container
│   │   ├── <container_1>.dsl.     <- " container container_1 'Container 1' {  "
│   │   │                             "     -> <container_N> '' 'REST'.        "
│   │   │                             " }                                      "
│   │   ├── <container_2>.dsl.           
│   │   └── <container_N>.dsl
│   └── software_system.dsl        <- " !include container "
```

Is there a way to bypass it because, basically, at the stage of building the model, it is non-critical? Or the only way is to place all relationships into the  `software_system.dsl` or a separate file?

**Comment by simonbrowndotje** [comment, marked as answer]
Lines are processed in order (https://github.com/structurizr/dsl/blob/master/docs/language-reference.md#general-rules), so you'll need to order your element and relationship definitions appropriately.

**Comment by okamiraner** [comment]
Well, if the connectivity looks like a complete graph (many omnidirectional relations), it wouldn't be much help...
